src/adapter/src/catalog/open.rs

```diff
@@ -264,24 +252,16 @@ impl Catalog {
 
             let is_read_only = storage.is_read_only();
             let mut txn = storage.transaction().await?;
-            // Choose a time at which to boot. This is the time at which we will run
-            // internal migrations.
-            //
-            // This time is usually the current system time, but with protection
-            // against backwards time jumps, even across restarts.
-            let boot_ts = {
-                let boot_ts = catalog_oracle::monotonic_now(
-                    config.now.clone(),
-                    previous_ts,
-                );
-                info!(%previous_ts, %boot_ts, "initialize_state");
-                if !is_read_only {
-                    // IMPORTANT: we durably record the new timestamp before using it.
-                    txn.set_timestamp(Timeline::EpochMilliseconds, boot_ts)?;
-                }
 
-                boot_ts
-            };
+            if !is_read_only {
+                // WIP: @joe, we're only persisting a boot_ts for the Catalog
+                // oracle, not for the Postgres oracle. We do it later though,
+                // when initializing the postgres oracle with an initial
+                // timestamp based on the boot_ts. It's not atomic with any
+                // catalog migrations and the like.
+                // IMPORTANT: we durably record the new timestamp before using it.
+                txn.set_timestamp(Timeline::EpochMilliseconds, boot_ts)?;
```

Comment:
It does seem wrong to me that we set only the catalog timestamp without even checking which implementation to use.